### ISSUE TITLE
Use a group0 timestamp for TRUNCATE

### ISSUE THREAD

**Issue by kostja** [scylladb/scylladb#18794]
In order to coordinate TRUNCATE with other DDL, as well as TRUNCATE across tables and derived views, we need to make sure it uses a single timestamp issued by the group0 leader and stored in the raft log.

**Comment by kbr-scylla** [comment]
Question to resolve: what to do with local (system) tables.

Some facts to consider when thinking about that question:
- currently, truncate of local table actually truncates it globally (IIRC) -- but not sure if it's intended, or just an artifact of the naive implementation which does not consider replication strategy during truncation, simply sends RPC to all nodes in the cluster
- local/system tables might contain data local to this node (that no other node contains or should contain)
- but local/system tables could also be modified with group 0 commands (which get applied on every node)
- we have tables (system.scylla_local) that contain both local only data and group0-committed data

I guess those last points are solvable by ensuring that tables contain either only local data, or only group0-committed data, never both.

**Comment by kbr-scylla** [comment]
@tgrabiec @bhalevy please comment on this issue to help me assess its importance
- in particular, is it a blocker for some tablets-related work?

For now I put P3 label on it, but I should be easily convinced to increase it to P2

**Comment by nodep** [comment]
Yes, I am working on #11230 The last few comments in the issue explain the plan of implementing TRUNCATE with tablets.

**Comment by kbr-scylla** [comment]
> Removing from plan until the confusion is resolved (see my comment here: https://github.com/scylladb/scylladb/issues/11230#issuecomment-2125039989)

Marking as `blocked` until someone explains to me what's happening and who owns this

**Comment by kbr-scylla** [comment]
From @bhalevy's comment https://github.com/scylladb/scylladb/issues/11230#issuecomment-2156326458 here's a summary of the situation as I understand it:
- to make TRUNCATE work correctly with tablets, we don't need this. There's a separate issue https://github.com/scylladb/scylladb/issues/16411 and it will have a different fix
- this issue actually consists of two parts: making truncate based on a timestamp, and distributing the timestamp through group 0. Generally people like the timestamp idea. But whether it should be distributed through group 0 is still debated.

For now there is no immediate need to work on this issue, if at all.

**Comment by mykaul** [comment]
@nodep - is this still relevant with your latest implementation?

**Comment by nodep** [comment]
> [@nodep](https://github.com/nodep) - is this still relevant with your latest implementation?

Yes, it is still relevant. My implementation does not currently propagate the truncate timestamp over group0.
I will look into it and see if I can extend the implementation to cover this, too.
